### PR TITLE
Added support for HackRF One

### PR DESCRIPTION
**Pull request by kr105**
I have added support for HackRF One device using libhackrf. I tested all the new functions successfully for a few days.

### CODE REVIEW

**Comment by cyclops1982** [comment]
I tested this and it works!

File: sdr_hackrf.c [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+        return false;
+    }
+
+    status = hackrf_set_vga_gain(HackRF.device, HackRF.vga_gain);
+    if (status != 0) {
+        fprintf(stderr, "HackRF: hackrf_set_vga_gain failed with code %d", status);
+        hackrf_close(HackRF.device);
+        hackrf_exit();
+        return false;
+    }
```
Comment (mutability) [inline sdr_hackrf.c:185]: Missing trailing \n on these error messages

File: sdr_hackrf.c [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+        fprintf(stderr, "HackRF: can't initialize sample converter\n");
+        return false;
+    }
+
+    return true;
+}
+
+struct timespec thread_cpu;
+
+int handle_hackrf_samples(hackrf_transfer *transfer)
```
Comment (mutability) [inline sdr_hackrf.c:203]: This needs to become static now that we're building with `-Wmissing-declarations` (that change postdates your PR)

File: sdr_hackrf.c [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+        return false;
+    }
+
+    status = hackrf_open(&HackRF.device);
+    if (status != 0) {
+        fprintf(stderr, "HackRF: hackrf_open failed with code %d", status);
+        hackrf_close(HackRF.device);
+        hackrf_exit();
+        return false;
+    }
```
Comment (mutability) [inline sdr_hackrf.c:145]: If I understand the hackrf API correctly, the cleanup paths here need to be a little more careful (don't call exit if init failed; don't call close if open failed)

**Pull request merged**